test: add initial type benchmarks

ssalbdivad opened the pull request:
This PR uses [`@ark/attest`](https://github.com/arktypeio/arktype/tree/main/ark/attest#benches) to add type-level benchmarks for several schemas + usage patterns in `packages/cli/src/__tests__/benches`.

The benchmarks can be run simply by executing the associated TS file (or through the `package.json` scripts in each bench dir that do the same).

Initial results were extremely good- linear scaling across all tested APIs.

Though client extensions add some general overhead, it along with the other APIs tested here should be able to scale quite far before type performance becomes an issue.

Most importantly, these benchmarks server as a pattern to adapt to other APIs + patterns that could cause problematic type performance in the future.

If anyone is aware of a relatively minimal repro (think ~10 models and <=50 ops) for bad type performance, I'd love to take a look. Otherwise, it should be easy enough to add them here in the future and start testing!

ssalbdivad committed:
begin integrating type benches

ssalbdivad committed:
remove dupes

ssalbdivad committed:
add logs.bench.ts

ssalbdivad committed:
improve xops bench

ssalbdivad committed:
cleanup bench

ssalbdivad committed:
move bench

ssalbdivad committed:
more bench cleanup

ssalbdivad committed:
more bench improvements

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/prisma/prisma?pullRequest=26894) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/prisma/prisma?pullRequest=26894) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/prisma/prisma?pullRequest=26894) it.</sub>

ssalbdivad committed:
remove unused client script

ssalbdivad committed:
Merge branch 'main' into tsperf

ssalbdivad committed:
merge main

prisma-bot committed:
chore(deps): update engines to 6.7.0-22.4c3ddfb7dad39802dcdfa9953a1bdf747a5494fa (#26948)

The base branch for this PR is: main
This automatic PR updates the engines to version
`6.7.0-22.4c3ddfb7dad39802dcdfa9953a1bdf747a5494fa`.
This will get automatically merged if all the tests pass.
:warning: If this PR needs to be updated, first remove the `automerge`
label before pushing to avoid automerge to merge without waiting for
tests.
| Package | NPM URL |
|---------|---------|
|`@prisma/engines-version`|
https://npmjs.com/package/@prisma/engines-version/v/6.7.0-22.4c3ddfb7dad39802dcdfa9953a1bdf747a5494fa|
|`@prisma/prisma-schema-wasm`|
https://npmjs.com/package/@prisma/prisma-schema-wasm/v/6.7.0-22.4c3ddfb7dad39802dcdfa9953a1bdf747a5494fa|
|`@prisma/query-engine-wasm`|
https://npmjs.com/package/@prisma/query-engine-wasm/v/6.7.0-22.4c3ddfb7dad39802dcdfa9953a1bdf747a5494fa|
|`@prisma/query-compiler-wasm`|
https://npmjs.com/package/@prisma/query-compiler-wasm/v/6.7.0-22.4c3ddfb7dad39802dcdfa9953a1bdf747a5494fa|
|`@prisma/schema-engine-wasm`|
https://npmjs.com/package/@prisma/schema-engine-wasm/v/6.7.0-22.4c3ddfb7dad39802dcdfa9953a1bdf747a5494fa|

[`prisma/prisma-engines@4c3ddfb7dad39802dcdfa9953a1bdf747a5494fa`](https://github.com/prisma/prisma-engines/commit/4c3ddfb7dad39802dcdfa9953a1bdf747a5494fa)

FGoessler committed:
feat: move type benchmarks to own directory and add docu and running utils

FGoessler committed:
chore: add to github actions CI

FGoessler committed:
Merge branch 'main' into tsperf

FGoessler committed:
fix: fix typo in dir name in CI

FGoessler committed:
chore: exclude type benchmarks from performance benchmarks CI

FGoessler committed:
chore: fix typecheck issues due to missing gen files on typecheck CI

FGoessler committed:
Merge branch 'main' into tsperf
